Assets/Content/Items/Consumables/Food/Consumable.cs

```diff
@@ -0,0 +1,200 @@
+    {
+        List<IInteraction> list = base.GenerateInteractions(interactionEvent).ToList();
+     
+        // Adds the interaction if it's not playing
+        if (CanConsume())
+        {
+            list.Add(new ConsumeInteraction {icon = sprite, verb = consumeVerb});
+        }
+
+        return list.ToArray();
+    }
+
+    public override void CreateInteractions(IInteractionTarget[] targets, List<InteractionEntry> interactions)
+    {
+        base.CreateInteractions(targets, interactions);
+        
+        // Adds the interaction if it's not playing
+        if (CanConsume())
+        {
+            interactions.Add(new InteractionEntry(targets[0],new ConsumeInteraction {icon = sprite, verb = consumeVerb}));
+        }
+    }
+
+    private class ConsumeInteraction : IInteraction
+    {
+        public Sprite icon;
+        public string verb;
+
+        public IClientInteraction CreateClient(InteractionEvent interactionEvent)
+        {
+            return null;
+        }
+
+        public string GetName(InteractionEvent interactionEvent)
+        {
+            return verb;
+        }
+
+        public Sprite GetIcon(InteractionEvent interactionEvent)
+        {
+            return icon;
+        }
+
+        public bool CanInteract(InteractionEvent interactionEvent)
+        {
+            // easy access to shit
+            GameObject source = interactionEvent.Source?.GetComponentInTree<Creature>().gameObject;
+            GameObject target = interactionEvent.Target.GetComponent<Transform>().gameObject; 
+            
+            // you can only interact with consumables or creatures
+            if (target.GetComponent(typeof(Consumable)) || source.GetComponent(typeof(Creature)))
+            {
+                if (!InteractionExtensions.RangeCheck(interactionEvent))
+                {
+                    return false;
+                }
+
+                return true;
+            }
+
+            return false;
+        }
+
+        public bool Start(InteractionEvent interactionEvent, InteractionReference reference)
+        {
+            GameObject source = interactionEvent.Source?.GetComponentInTree<Creature>().gameObject;
+            GameObject target = interactionEvent.Target.GetComponent<Transform>().gameObject;            
+            Consumable itemInHand = source?.GetComponentInChildren<Hands>().GetItemInHand()?.GetComponent<Consumable>();
+	        
+			//Debug.Log("source:  " + source.name + " target: " + target?.name + " item: " + itemInHand?.gameObject.name);
+			
+            // Item in hand and interacting with origin
+            if (target == null) 
+            {
+                itemInHand.ConsumeAction(source);
+            }
+            // Item not in hand
+            if (target.GetComponent(typeof(Consumable)))
+            {
+                target.GetComponent<IConsumable>().ConsumeAction(source);
+            }
```

Comment:
When checking for components, always assign them to a variable to avoid multiple calls to GetComponent.
Also check for IConsumable in the if statement, not Consumable.